tests/e2e/main_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"flag"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"testing"
+)
+
+const (
+	epHealthz = "/healthz"
+	epMetrics = "/metrics"
+)
+
+var (
+	framework *Framework
+)
+
+type Framework struct {
+	KsmClient *KSMClient
+}
+
+func NewFramework(ksmurl string) (*Framework, error) {
+	ksmClient, err := NewKSMClient(ksmurl)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Framework{
+		KsmClient: ksmClient,
+	}, nil
+}
+
+type KSMClient struct {
+	endpoint *url.URL
+	client   *http.Client
+}
+
+func NewKSMClient(address string) (*KSMClient, error) {
+	u, err := url.Parse(address)
+	if err != nil {
+		return nil, err
+	}
+	u.Path = strings.TrimRight(u.Path, "/")
+
+	return &KSMClient{
+		endpoint: u,
+		client:   &http.Client{},
+	}, nil
+}
+
+func (k *KSMClient) Healthz() (bool, error) {
```

Comment:
I would call this maybe `isHelathz()` for lack of better ideas, also this is not used outside of this file so lets not expose it.